src/consumer.ts

```diff
@@ -63,11 +63,14 @@ export class Consumer extends TypedEventEmitter {
   private authenticationErrorTimeout: number;
   private pollingWaitTimeMs: number;
   private pollingCompleteWaitTimeMs: number;
+  private concurrencyWaitTimeMs: number;
   private heartbeatInterval: number;
   private isPolling = false;
   private stopRequestedAtTimestamp: number;
   public abortController: AbortController;
   private extendedAWSErrors: boolean;
+  private concurrency: number;
```

Comment:
We may want a maxConcurrency setting that ensures we're not trying to do silly stuff.